.github/workflows/inspektor-gadget.yml

```diff
@@ -697,6 +697,53 @@ jobs:
           rm -rf /tmp/.buildx-cache
           mv /tmp/.buildx-cache-new /tmp/.buildx-cache
 
+  analyze-ig-memory:
+    name: IG binary memory analysis
+    runs-on: ubuntu-latest
+    needs: 
+      - build-ig 
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@3041bf56c941b39c61721a86cd11f3bb1338122a # v5.2.0
+        with:
+          go-version: '1.23.4'
+          cache: true
+        id: go
+      - name: Get ig-linux-amd64.tar.gz from artifact.
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: ig-linux-amd64-tar-gz
+          path: /home/runner/work/inspektor-gadget/
+      - name: Unpack ig-linux-amd64.tar.gz
+        run: |
+          tar zxvf /home/runner/work/inspektor-gadget/ig-linux-amd64.tar.gz
+      - name: download pprof
+        run: |
+          go install github.com/google/pprof@latest
+      - name: Run ig gadget & wait for 10 seconds
+        run:
+          sudo ./ig run trace_dns:latest --verify-image=false --pprof-addr=localhost:6060 &
+          sleep 10
+      - name: Run pprof
+        run: |
+          curl http://localhost:6060/debug/pprof/heap > heap.out
+      - name: Analyze memory
+        run: |
+          go tool pprof -top -inuse_space ./ig heap.out > inuse_space.txt
+          go tool pprof -top -alloc_space ./ig heap.out > alloc_space.txt
```

Comment:
Similarly to the other PR, the commands from "tar zxvf" until here could be a single step.